diff --git a/RATL.diff b/RATL.diff
index 845942b..e69de29 100644
--- a/RATL.diff
+++ b/RATL.diff
@@ -1,51 +0,0 @@
-diff --git a/include/atl/atlsimpstr.h b/include/atl/atlsimpstr.h
-index 7d78f41..a2956a9 100644
---- a/include/atl/atlsimpstr.h
-+++ b/include/atl/atlsimpstr.h
-@@ -6,6 +6,12 @@
- #include <atlcore.h>
- #include <atlexcept.h>
- 
-+#ifdef __RATL__
-+    #ifndef _In_count_
-+        #define _In_count_(nLength)
-+    #endif
-+#endif
-+
- namespace ATL
- {
- struct CStringData;
-diff --git a/include/atl/atlwin.h b/include/atl/atlwin.h
-index b970fe7..533114b 100644
---- a/include/atl/atlwin.h
-+++ b/include/atl/atlwin.h
-@@ -49,6 +49,12 @@ inline LONG_PTR GetWindowLongPtr(HWND hWnd, int nIndex)
- #pragma push_macro("SubclassWindow")
- #undef SubclassWindow
- 
-+#ifdef __RATL__
-+    #ifndef _Post_z_
-+        #define _Post_z_
-+    #endif
-+#endif
-+
- namespace ATL
- {
- 
-diff --git a/include/wine/test.h b/include/wine/test.h
-index 849438f..55a6c87 100644
---- a/include/wine/test.h
-+++ b/include/wine/test.h
-@@ -23,8 +23,12 @@
- 
- #include <stdarg.h>
- #include <stdlib.h>
-+#ifdef __RATL__
-+#include <windows.h>
-+#else
- #include <windef.h>
- #include <winbase.h>
-+#endif
- 
- #ifdef __WINE_CONFIG_H
- #error config.h should not be used in Wine tests
diff --git a/atltests/CComQIPtr.cpp b/atltests/CComQIPtr.cpp
index 1ab179b..35cfedf 100644
--- a/atltests/CComQIPtr.cpp
+++ b/atltests/CComQIPtr.cpp
@@ -76,7 +76,7 @@ public:
 };
 
 // Yes this sucks, but we have to support GCC. (CORE-12710)
-#ifdef __REACTOS__
+#if defined(__REACTOS__) || defined(__RATL__)
 #define DECLARE_QIPTR(type)     CComQIIDPtr<I_ID(type)>
 #elif defined(__GNUC__)
 #define DECLARE_QIPTR(type)     CComQIIDPtr<I_ID(type)>
diff --git a/include/atl/atlsimpstr.h b/include/atl/atlsimpstr.h
index 7d78f41..a2956a9 100644
--- a/include/atl/atlsimpstr.h
+++ b/include/atl/atlsimpstr.h
@@ -6,6 +6,12 @@
 #include <atlcore.h>
 #include <atlexcept.h>
 
+#ifdef __RATL__
+    #ifndef _In_count_
+        #define _In_count_(nLength)
+    #endif
+#endif
+
 namespace ATL
 {
 struct CStringData;
diff --git a/include/atl/atlwin.h b/include/atl/atlwin.h
index b970fe7..533114b 100644
--- a/include/atl/atlwin.h
+++ b/include/atl/atlwin.h
@@ -49,6 +49,12 @@ inline LONG_PTR GetWindowLongPtr(HWND hWnd, int nIndex)
 #pragma push_macro("SubclassWindow")
 #undef SubclassWindow
 
+#ifdef __RATL__
+    #ifndef _Post_z_
+        #define _Post_z_
+    #endif
+#endif
+
 namespace ATL
 {
 
diff --git a/include/wine/test.h b/include/wine/test.h
index 0cbc400..231e117 100644
--- a/include/wine/test.h
+++ b/include/wine/test.h
@@ -23,8 +23,12 @@
 
 #include <stdarg.h>
 #include <stdlib.h>
+#ifdef __RATL__
+#include <windows.h>
+#else
 #include <windef.h>
 #include <winbase.h>
+#endif
 
 #ifdef __WINE_CONFIG_H
 #error config.h should not be used in Wine tests
